libp2p/switch.nim

```diff
@@ -250,6 +231,33 @@ proc stop*(s: Switch) {.async.} =
 
   trace "Switch stopped"
 
+proc start*(s: Switch): Future[seq[Future[void]]] {.async, gcsafe.} =
+  trace "starting switch for peer", peerInfo = s.peerInfo
+  var startFuts: seq[Future[void]]
+  for t in s.transports: # for each transport
+    for i, a in s.peerInfo.addrs:
+      if t.handles(a): # check if it handles the multiaddr
+        let transpStart = t.start(a)
+        startFuts.add(transpStart)
+        try:
+          await transpStart
+          s.peerInfo.addrs[i] = t.ma # update peer's address
+          s.acceptFuts.add(s.accept(t))
+        except CancelledError:
+          await s.stop()
```

Comment:
technically, this will call stop on transports which have not yet started .. that's an imbalance in starts vs stops, but one can hope that's handled somewhere..

Reply:
Yes, stopping a non started transport is not an issue